### PR TITLE
Clean up env and fix url

### PR DESCRIPTION
**Pull request by shamsher31**

### CODE REVIEW

File: pkg/test/env/istio.go [hunk 1]
```diff
@@ -95,9 +95,6 @@ var (
 	// ChartsDir is the Kubernetes Helm chart directory in the repository
 	ChartsDir = path.Join(IstioSrc, "install/kubernetes/helm")
```
Comment (shamsher31) [inline pkg/test/env/istio.go:96]: Note that `install/kubernetes/helm` is also not valid path from `1.6`. `ChartsDir` is referenced in [ServiceAccountFilePath](https://github.com/istio/istio/pull/24174/files#diff-b7bc2684a03df2e326b38e1c6d0a7624R105) which also did not exists.

File: operator/ARCHITECTURE.md [hunk 2]
```diff
@@ -24,17 +24,17 @@ run a privileged controller in the cluster.
...
 ## Terminology
 
 Throughout the document, the following terms are used:
 
 - `IstioOperatorSpec`: The API directly defined in the
-[IstioOperatorSpec proto](https://github.com/istio/api/mesh/v1alpha1/operator.proto),
+[IstioOperatorSpec proto](https://github.com/istio/api/blob/master/operator/v1alpha1/operator.proto),
 including feature and component groupings, namespaces and enablement, and per-component K8s settings.
 - Helm values.yaml API, implicitly defined through the various values.yaml files in the
 [Helm charts](https://github.com/istio/installer) and schematized in the operator through
```
Comment (morvencao) [inline operator/ARCHITECTURE.md:40]: The url `https://github.com/istio/installer` here is also deprecated.

Comment (shamsher31) [reply operator/ARCHITECTURE.md]: Fixed, Thanks

File: pkg/test/env/istio.go (hunk 1 above)
Comment (howardjohn) [reply pkg/test/env/istio.go:96]: I noticed this previously. Looks like the only test using it is skipped so it doesn't fail. I recommend we just wait for 1.7 when mixer is gone then we can just delete it all?

Comment (shamsher31) [reply pkg/test/env/istio.go:96]: Sounds good, I think deleting `IstioChartDir` is ok with this PR as it is not referenced anywhere in the code.

**Comment by shamsher31** [comment]
@howardjohn @morvencao can you PTAL ?

File: pkg/test/env/istio.go (hunk 1 above)
Comment (morvencao) [reply pkg/test/env/istio.go:96]: Not sure if it is safe to remove `IstioChartDir`.
Leave this to @howardjohn

**Review by howardjohn — APPROVED** [review]

**Pull request merged**